emcc.py

```diff
@@ -2385,6 +2385,9 @@ def check_memory_setting(setting):
         exit_with_error('EXPORT_ES6 and ENVIRONMENT=*node* requires USE_ES6_IMPORT_META to be set')
       settings.USE_ES6_IMPORT_META = 1
 
+  if 'EXPORT_NAME' in user_settings and not settings.MODULARIZE:
+    exit_with_error('EXPORT_NAME requires MODULARIZE to be set')
```

Comment:
I assume this is a breaking change? How likely is it to trip someone up?

Reply:
Hmm also, what name do we use for non-MODULARIZE build? (for the variable)

I haven't used it in ages, but I thought both names would be under the same setting?

Reply:
For non-MODULARIZE builds the global `Module` always exists.      IIUC, `MODULARIZ`E is the only solution there is to not have a `Module` global.  

Another way of putting it: without `MODULARIZE` there is nothing to export from .. all emscripten globals live in the global JS scope and all emscripten variables can collide with other JS variables.  The only way to avoid those collisions (that I know of) is to use `MODULARIZE`.  

Having said all that it looks like `MINIMAL_RUNTIME` is using `EXPORT_NAME` even when `MODULARIZE` is not set, so we would need to resolve that before landing this change.